### PR TITLE
WT-8925 Switch S3 extension unit tests to Catch2 framework

### PR DESCRIPTION
**Pull request by ruby-oujo**
* Switch S3 extension unit tests to Catch2 framework.
* Made changes to the way Catch2 is imported by making Catch2 accessible earlier on in WiredTiger, so that it can be accessed by the s3 extension and allow it to have its own executable.
* Added a class `S3Cleanup`  with a destructor to clean up objects after a test. 
* Updated the documentation for S3 testing. 
* Updated the build for tiered tests in `evergreen.yml`.

### CODE REVIEW

File: ext/storage_sources/s3_store/test/test_s3_connection.cpp [hunk 1]
```diff
@@ -93,325 +81,143 @@ setupTestDefaults()
...
-
-    return (ret);
+    return (0);
 }
 
-// Lists S3 objects under the test bucket.
-// Todo: Remove code duplication in this function.
-int
-TestListObjects(const Aws::S3Crt::ClientConfiguration &config)
+TEST_CASE("Testing S3 Connection", "something")
```
Comment (Sean04) [inline ext/storage_sources/s3_store/test/test_s3_connection.cpp:90]: What does the "something" refer to here?

**Review by Sean04 — APPROVED** [review]
Nice work, this simplifies the unit test a lot! LGTM with 1 minor comment.

File: ext/storage_sources/s3_store/test/test_s3_connection.cpp (hunk 1 above)
Comment (ruby-oujo) [reply ext/storage_sources/s3_store/test/test_s3_connection.cpp:90]: Forgot to replace this line! Was just a filler till I decided what to label the tests. Will change this now.

**Review by sulabhM — APPROVED** [review]
I have some minor comments that wont need a re-review.
I really liked this change, LGTM!

I have left a comment on multithreading the catch2 test runs if possible. Feel free to ignore if that is a big ask. If you do make the tests multithreaded I will like to review again.

File: ext/storage_sources/s3_store/test/test_s3_connection.cpp [hunk 2]
```diff
@@ -93,325 +81,143 @@ setupTestDefaults()
...
-        return (TEST_FAILURE);
+    SECTION("Check if object exists in S3, and if object size is correct)", "[s3-connection]")
+    {
+        REQUIRE(conn.ObjectExists(objectName, exists, objectSize) == 0);
+        REQUIRE(objectSize == 0);
+        REQUIRE(conn.PutObject(objectName, fileName) == 0);
+        REQUIRE(conn.ObjectExists(objectName, exists, objectSize) == 0);
+        REQUIRE(exists);
+        REQUIRE(objectSize == payload.length());
+        REQUIRE(conn.DeleteObject(objectName) == 0);
```
[the remark this answers is not in the document]
Comment (sulabhM) [reply ext/storage_sources/s3_store/test/test_s3_connection.cpp:123]: ```Catch2 keeps test execution in one process strictly serial, and there are no plans to change this. If you find yourself with a test suite that takes too long to run and yo uwant to make it parallel, you have to run multiple processes side by side.```

I stand corrected. Catch2 wont run test cases in parallel unless split between separate binaries. So please ignore my above comment.

**Review by jeremythorp — APPROVED** [review]
LGTM too! Nice job, and the code is now so much shorter and simpler!

**Pull request merged**